Closure_Front_End/src/Utils.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * check if the status code of an HTTP response is 'Ok'.
+ * @param statusCode - the status code of an HTTP response
+ * @returns {boolean|boolean} 'true' if the status code is an 'Ok' status code. 'false' otherwise.
+ */
+function validateStatusCode (statusCode) {
+    return statusCode >= 200 && statusCode < 400
+}
+
+/**
+ * pops out an alert with generic error message. We call it when there was an error while performing an HTTP call.
+ */
+function popOutStatusCodeError(){
+    alert("Error occurred. Couldn't perform the operation.")
```

Comment:
Alerts aren't a very good way of showing errors:

- We should use a proper Vue component for displaying a "toast", which could be used for many kinds of errors, not just network related errors. Something that looks like one of the error notifications [here](https://bulma.io/documentation/elements/notification/)

- The error message should be informative(and in Hebrew, at least some of it)
  For example, error codes that begin with 500 indicate a problem in our server's code
  Error code 404 might indicate that the server is down
  A network error might happen if the client's internet is down(in that case, there wouldn't be any HTTP status code)

Reply:
why alert isn't good?

Reply:
`console.alert` doesn't provide a good user experience, see [here](https://stackoverflow.com/questions/8825384/alert-is-bad-really) and [here](https://www.quora.com/Why-is-JavaScripts-alert-box-considered-a-bad-design-from-usability-point-of-view)

For example, suppose you're trying to search for courses and there are network errors. With every letter you type, you'll get an alert box

What about errors that don't deserve such a disruptive alert box? For example, maybe we'll implement caching of  courses/tracks within local-storage, so if the network requests fail, we'll get a more limited selection of courses, but we should still be able to proceed with the app

Or, how about auto-save: our app could auto-save the User courses into both local storage and to the DB. Without network access, the user should get a warning that his changes aren't saved to the DB but are still saved on the local-storage of his browser. 

Basically, in many scenarios where the user has no ways of dealing with the error(e.g, network errors), it is best to warn him about it in a non-intrusive manner (e.g, a "toast")

And in the cases where he can deal with it(business logic errors), then we'll probably want a nicer and more informative way of letting him deal with it.

Reply:
got it, thanks :) I will fix it